### PR TITLE
Add choice_loader options for multi choice field

### PR DESCRIPTION
**Pull request by Mangetsu**
Fix ModelType : replace choice_list by choice_loader for sf2.7+
Fix ModelsToArrayTransformer : add "instance of ModelChoiceLoader" test for multi-choice form widget to avoid Exception in sf2.7+

### CODE REVIEW

**Comment by Mangetsu** [comment]
@greg0ire i hope this is the last time i bother you with my code changes (next time kick me out please :laughing: )

**Comment by greg0ire** [comment]
Haha sorry, but it is still not good : 

```patch
index cc7b257..88fdf93 100644
--- a/Form/DataTransformer/ModelsToArrayTransformer.php
+++ b/Form/DataTransformer/ModelsToArrayTransformer.php
@@ -51,8 +51,8 @@ class ModelsToArrayTransformer implements DataTransformerInterface
      */
     public function __construct($choiceList, ModelManagerInterface $modelManager, $class)
     {
-        if (!$choiceList instanceof ModelChoiceList 
-            && !$choiceList instanceof ModelChoiceLoader 
+        if (!$choiceList instanceof ModelChoiceList
+            && !$choiceList instanceof ModelChoiceLoader
             && !$choiceList instanceof LazyChoiceList) {
             throw new RuntimeException('First param passed to ModelsToArrayTransformer should be instance of
                 ModelChoiceLoader or ModelChoiceList or LazyChoiceList');
```

Almost there!

**Comment by greg0ire** [comment]
Make sure you configure PHPStorm to remove trailing spaces ;)

**Comment by greg0ire** [comment]
http://www.craftitonline.com/2011/06/set-phpstorm-to-remove-trailing-spaces-on-lines/

**Comment by Mangetsu** [comment]
I made a joke of myself :laughing:

**Comment by Mangetsu** [comment]
Well I will re-work this tonight at home. sorry for disturbing you all day long.

**Comment by greg0ire** [comment]
> I made a joke of myself :laughing:

At least you try to contribute, which is far more than what most people do :)

**Comment by greg0ire** [comment]
> Well I will re-work this tonight at home. sorry for disturbing you all day long.

Ok, see you :)

**Comment by Mangetsu** [comment]
@greg0ire : I've done all you asked, seems SonataCI is not passing my changes.

File: Form/ChoiceList/ModelChoiceList.php [hunk 1]
```diff
@@ -97,10 +97,10 @@ class ModelChoiceList extends SimpleChoiceList
...
     public function __construct(ModelManagerInterface $modelManager, $class, $property = null, $query = null, $choices = array(), PropertyAccessorInterface $propertyAccessor = null)
     {
-        $this->modelManager   = $modelManager;
-        $this->class          = $class;
-        $this->query          = $query;
-        $this->identifier     = $this->modelManager->getIdentifierFieldNames($this->class);
+        $this->modelManager = $modelManager;
+        $this->class = $class;
+        $this->query = $query;
+        $this->identifier = $this->modelManager->getIdentifierFieldNames($this->class);
```
Comment (greg0ire) [inline Form/ChoiceList/ModelChoiceList.php]: Errors are about this. Why did you change this?

Comment (Mangetsu) [inline Form/ChoiceList/ModelChoiceList.php]: I didn't, it was php cs fixer that done some white space removing I guess.

Comment (greg0ire) [inline Form/ChoiceList/ModelChoiceList.php]: Wow strange. It is supposed to be configured not to.

**Comment by greg0ire** [comment]
Why did you close your PR? You can update it with new commits by using `git push --force`, you know?

**Comment by Mangetsu** [comment]
I tried many time, I restart clean PR just to be sure everything is correctly done, but I keep getting errors now for phpdoc, I'm giving up ^^

**Comment by greg0ire** [comment]
> but I keep getting errors now for phpdoc, I'm giving up ^^

Nooooo! You were so close!
